astropy/timeseries/binned.py

```diff
@@ -326,8 +333,14 @@ def read(self, filename, time_bin_start_column=None, time_bin_end_column=None,
 
                 time_bin_end = None
 
-            return BinnedTimeSeries(data=table,
+            if time_bin_start.isscalar and time_bin_size.isscalar:
```

Comment:
I wonder if this condition is ever satisfied since `time_bin_start` and `time_bin_size` are read from columns with the start time and bin size for each bin; unless there is a single element in the table (?). If this condition is never true, it could be removed. I am not entirely sure about this so we will wait for a review.

Reply:
Whenever an exception is raised (or there's a need to call default table readers), the return statement always passes `n_bins` as a parameter. So in some cases (for example `function test_read()` in tests/test_binned.py and test case in astropy/docs/timeseries/io.rst ) the exception we created is raised. So either changing the test case or changing this seemed like the correct option.

In simple words, I checked that this condition is sometimes true for the given test cases, and removing it will lead to failure of the above-mentioned test cases.

Reply:
I agree that this change is required to fix issues with the test cases as well as more generally since we do not want to pass `n_bins = len(table)` when `time_bin_start` or `time_bin_end` is not scalar. My comment was whether we ever _need_ to pass `n_bins = len(table)`, that is, is the condition `if time_bin_start.isscalar and time_bin_size.isscalar` ever true in the case of reading a table whose columns specify `time_bin_start` and `time_bin_size`. The condition might _only_ be true when there is one element. However, I do not know enough about `timeseries` to say for sure. I will let @astrofrog decide 😄

Reply:
There may be better ways to address the root cause (why is n_bins passed in that case?), but I think this is an acceptable solution. All I'm asking for is to put a comment in the code to explain this (essentially what you said above: "When XXX happens, n_bins is passed, but that causes exception X, so here we do this ..."). 
After that, I'm happy to approve an merge.